README.md

```diff
@@ -1 +1,232 @@
+
+*onSuccess* It will:  
+  * Put `${action.target}Loading` in `false`  
+  * Put `${action.target}Error` in `null`  
+  * Fill `${action.target}` with your `action.payload` by default, or use a selector provided  
+  Examples:  
+    ```
+    const selector = (action, state) => action.payload || state[action.target];
+    const reducerDescription = {
+      'SUCCESS': onSuccess(),
+      'SUCCESS_AND_ADD': onSuccess(selector)
+    };
+    ```
+*onFailure*: It will:  
+  * Put `${action.target}Loading` in `false`  
+  * Put `${action.target}Error` with your `action.payload`. This also supports a selector.  
+*onLoading*: It will put `${action.target}Loading` in true. Also supports selectors.  
+*onLoaded*: It will put `${action.target}Loading` in false. Also supports selectors.  
+
+_Spoiler alert_: The last ones are useful to implement SUCCESS/FAILURE async actions pattern in thunk actions. Soon will be available this creator.  
+
+We are currently writting some other effects:  
+```
+onSetValue
+onReset
+onDelete
+onToggle
+onAppend
+onCycle
+onMap
+```
+
+And other utilities like  
+```
+composeEffects
+```
+
+New effects are welcome ! Feel free to open an issue or even a PR.  
+
+## Creators [WIP]  
+There are a few creators that also ease writing Redux reducers and actions.  
+
+*createReducer*: Receives an initialState and a reducer description. It replaces the traditional switch.  
+  Example:  
+  ```
+  const reducerDescription = {
+    'ACTION_NAME': (state, action) => ({ ...state, aTarget: ':)' })
+  }
+
+  const initialState = { aTarget: null };
+  const reducer = createReducer(initialState, reducerDescription);
+  ...
+  dispatch({ type: 'ACTION_NAME' }); // state.aTarget = ':)'
+  ```
+*createAction*: Receives a type, a target and optionally a payload. Returns a new action.  
+*createTypes*: Receives a string list and another string to prepend a namespace.  
+  Example:  
+  ```
+  const actions = createTypes(['ACTION1', 'ACTION2'], '@@NAMESPACE');
+  actions.ACTION1 === '@@NAMESPACE/ACTION1';
+  ```
+
+We are currently working on these creators:  
+```
+createThunkAction
+createMultiTargetAction
+createPollingAction
+```
+Since state handling is decoupled from its state, we could create some more complex async actions, or even map
+an effect with an action type to create families of actions.  
+More crazy and useful ideas are welcome too !  
+
+## Completers [WIP]
+You could use completers to reduce your code size.  
+
+*completeState*: Receives a state description and a list of target exceptions.  
+  Example:  
+  ```
+  const initialLongState = {
+    thing: null,
+    thingLoading: false,
+    thingError: null,
+    otherThing: null,
+    otherThingLoading: false,
+    otherThingError: null,
+    anotherThing: null
+  };
+
+  const initialStateDescription = {
+    thing: null,
+    otherThing: null,
+    anotherThing: null
+  }
+
+  const initialState = completeState(initialStateDescription, ['anotherThing']);
```

Comment:
shouldnt this be `const initialState = completeState(initialStateDescription, ['Error', 'Loading']);